core/src/main/scala/chisel3/internal/package.scala

```diff
@@ -180,8 +180,7 @@ package object internal {
         color
       case _ => return
     }
-    val enabledLayers = Builder.enabledLayers.view ++ Builder.layerStack.headOption
-    if (enabledLayers.exists(_.canWriteTo(destLayer))) {
+    if (Builder.layerStack.headOption.forall(_.canWriteTo(destLayer))) {
```

Comment:
Clever use of `forall`. 👍

Reply:
Revisiting, this is holdover from replacing the previous `exists` with `forall` -- but now that it's a single element (Option) it's not necessary but will leave as-is.

If a block could be colored by `A` AND `B` (where each could be activated independently), we'd want "all" but presently we can continue to just check the top of the stack.